### DISCUSSION TITLE
使用react开发的taro小程序报错问题

### DISCUSSION THREAD

**Discussion by sofiagogogo** [NervJS/taro#11250] [Q & A] [answered]
'View' cannot be used as a JSX component.\n  Its element type 'ReactElement<any, any> | Component<ViewProps, any, any> | null' is not a valid JSX element.“.项目运行没有问题，但是一直会有标签报错。已导入要使用的标签。
![image](https://user-images.githubusercontent.com/50742189/153144642-cf436ae7-be4b-4975-ba07-dccdefe57014.png)

**Comment by canyueduhong** [comment]
不知道我遇到的和你的是否一样，在vscode中也会有相同的问题。不是因为小程序的原因，是因为tsconfig.json设置了"jsx": "react",他会检查在当前文件的react是否导入。可以设置为"jsx": "react-jsx"。或在报错页面导入react：import React from "react";

**Reply by sofiagogogo** [reply]
tsconfig.ts文件的确设置"jsx": "react-jsx"。但每个页面都有导入react,报错一直存在。

**Reply by CrazyMan2** [reply]
我也是，请问怎么解决

**Reply by Chen-jj** [reply]
好像和 #11097 是同一个问题，可以参阅一下。其在 3.4.2 已修复

**Reply by sofiagogogo** [reply]
> 好像和 #11097 是同一个问题，可以参阅一下。其在 3.4.2 已修复

好的，感谢。具有参考意义。

**Reply by sofiagogogo** [reply]
> 我也是，请问怎么解决

参考[#11097](https://github.com/NervJS/taro/issues/11097) 这个issue，升级了package.json里面的依赖，然后删除了node_modules文件夹，使用yarn(代替npm install)重新安装就可以了。不过好像导致了一个另外一个引用react is not a module，但比之前代码看起来好多了，不是一片红色报错。

**Comment by sofiagogogo** [comment, marked as answer]
参考#11097 这个issue，升级了package.json里面的依赖，然后删除了node_modules文件夹，使用yarn(代替npm install)重新安装就可以了。不过好像导致了一个另外一个引用react is not a module，但比之前代码看起来好多了，不是一片红色报错。

**Comment by panyu97py** [comment]
还有一种情况 yarn 会在这个目录下 安装一个 @types/react 的包 版本不一致就会报错 删除后可解决
/Users/[your username]/node_modules/@types
